sdk/src/main/scala/com/horizen/certificatesubmitter/CertificateSubmitter.scala

```diff
@@ -80,30 +90,91 @@ class CertificateSubmitter
+      sidechainWallet.secret(publicKey).map(_.sign(message).bytes).getOrElse(missedPrivateKeySignatureBytesPlaceholder)
+    }
+
+    BackwardTransferLoader.schnorrFunctions.createProof(
+      withdrawalRequestBoxes.asJava,
+      endWithdrawalEpochBlockHash,
+      prevEndWithdrawalEpochBlockHash,
+      schnorrSignatures.asJava,
+      publicKeysBytes.asJava,
+      params.backwardTransferThreshold,
+      params.provingKeyFilePath)
+  }
+
+  protected def trySubmitCertificate: Receive = {
+    case SemanticallySuccessfulModifier(_: SidechainBlock) => {
+      val submitCertificateFunction: View => SubmitResult = submitCertificate
 
-    val submitMessage: SubmitMessageType =
-      GetDataFromCurrentView[SidechainHistory, SidechainState, SidechainWallet, SidechainMemoryPool, SubmitResult](submitCertificateFunction)
+      val submitMessage: SubmitMessageType =
+        GetDataFromCurrentView[SidechainHistory, SidechainState, SidechainWallet, SidechainMemoryPool, SubmitResult](submitCertificateFunction)
 
-    val forgedBlockAsFuture = (sidechainNodeViewHolderRef ? submitMessage).asInstanceOf[Future[SubmitResult]]
-    forgedBlockAsFuture.onComplete{
-      case Success(SubmitSuccess) =>
-        log.info(s"Backward transfer certificate was successfully created.")
+      val certificateSubmittingResult = (sidechainNodeViewHolderRef ? submitMessage).asInstanceOf[Future[SubmitResult]]
+      certificateSubmittingResult.onComplete{
+        case Success(SubmitSuccess) =>
+          log.info(s"Backward transfer certificate was successfully created.")
 
-      case Success(SubmitFailed(ex)) =>
-        log.info("Creation of backward transfer certificate had been failed." + ex)
+        case Success(SubmitFailed(ex)) =>
+          log.info("Creation of backward transfer certificate had been failed." + ex)
 
-      case Failure(ex) =>
-        log.info("Creation of backward transfer certificate had been failed. " + ex)
+        case Failure(ex) =>
+          log.info("Creation of backward transfer certificate had been failed. " + ex)
+      }
     }
   }
 
-  override def receive: Receive = {
-    case SemanticallySuccessfulModifier(sidechainBlock: SidechainBlock) => {
-      trySubmitCertificate
+  protected def checkSubmitter: Receive = {
+    case SidechainAppEvents.SidechainApplicationStart => {
+      val submitterCheckingFunction: View => Unit = checkSubmitterMessage
+
+      val checkMessage: CheckMessageType =
+        GetDataFromCurrentView[SidechainHistory, SidechainState, SidechainWallet, SidechainMemoryPool, Unit](submitterCheckingFunction)
+
+      val checkAsFuture = (sidechainNodeViewHolderRef ? checkMessage).asInstanceOf[Future[Unit]]
+      checkAsFuture.onComplete{
+        case Success(()) =>
+          log.info(s"Backward transfer certificate submitter was successfully started.")
+
+        case Failure(ex) =>
+          log.info("Backward transfer certificate submitter failed to start due:" + ex)
+          throw ex
```

Comment:
is it ok to throw exception here?
Can it be possible to not to have an access to nodeviewholder during submitter creation?

Reply:
we check it after whole application start, thus we have that access. Discuss: what to do in case if submitter is "corrupted"